### ISSUE TITLE
Information about builds missing in Home App, Builder's list of completed builds 

### ISSUE THREAD

**Issue by YngveNPettersen** [buildbot/buildbot#4516]
When looking at the builds in the Home App and for the individual builders (e.g. URL #/builders/1 ) it is not possible to tell what the build is in your own environment, and it is missing other information that can be considered important.

At present the Home App only displays the buildername/1 name, build status, and duration of the build. The builder page displays the status, how long ago the build completed, it's duration, an owner name, the worker name.

In our environment, while some of the information is useful (such as duration, status) the following information is better suited to our purposes:

- The date and time the build completed
- The Git commit ID
- The branch name
- The build version number 1.2.3.4 generated for each build

Other projects might have other needs.

AFAICT it is not possible to customize the home app and builder pages to display such information

BTW, the status information for failures display the first X characters of the shell command that failed, which is not very useful in most cases (e.g. "python -u", which is the start of dozens of commands used in a single build is all that is displayed for many of our builds); the failed step's name would be vastly more informative.

**Comment by YngveNPettersen** [comment]
My thought on how this could be implemented is to be able to configure the fields selected in the case of the builder pages, with the display done by util.Property or util.Interpolate&co, and something similar for the Home App.

The main problem for me, regarding dashboards is that while I am mainly a C++ dev, and I am able to do quite a bit in Python and the Flask system, I am not a JS dev (I am having a colleague do the JS and HTML customization in a couple of Flask apps we are developing), and in the case of Buildbot and Angular&co, IMO the documentation is not remotely sufficient for me to actually develop even a basic app and figure out how to get our buildbot to actually use it, especially without creating our own fork of the code (which is something I want to avoid). Some very detailed step by step basic examples would help a lot (e.g. a more advanced Framboozle example)

**Comment by tardyp** [comment]
There is a 1h30 video associated to the blog post. I don't think I can do more step by step :)

**Comment by YngveNPettersen** [comment]
I prefer being able to _read_ documentation and examples, in association with the rest of the documentation, especially when developing my own version; going back an forth in a video does not do much for being wanting to correlate many different aspects of the information. Also, third party hosted content may suddenly be transferred to the the great big archive.

**Comment by YngveNPettersen** [comment]
I have just pushed up the modifications I did to get our Base App to present the details we want. The uploaded patch is an example, not a patch.

What I would like is to be able to do is to configure this kind of modifications from the master.cfg file in the c["www"] object.

https://github.com/YngveNPettersen/buildbot/commit/3487c85e598626eb9b4b83bfc0d59537c1e172eb

BTW, this patch also increases the number of builds to load in the Home page to 100, a number that I would also like to configure from the master.cfg
